wandb/sdk_py27/lib/file_stream_utils.py

```diff
@@ -25,35 +25,51 @@ def _str_size(x):
     def _file_size(file):
         size = file.get("_size")
         if size is None:
-            size = sum(map(_str_size, file["content"]))
+            size = 0
+            content = file["content"]
+            for i in range(file["_start"], file["_end"]):
+                size += _str_size(content[i])
             file["_size"] = size
         return size
 
     def _split_file(file, num_lines):
-        offset = file["offset"]
-        content = file["content"]
-        name = file["name"]
-        f1 = {"offset": offset, "content": content[:num_lines], "name": name}
-        f2 = {
-            "offset": offset + num_lines,
-            "content": content[num_lines:],
-            "name": name,
-        }
+        f1 = file.copy()
+        f1["_end"] = f1["_start"] + num_lines
+        f2 = file.copy()
+        f2["_start"] += num_lines
+        f2["offset"] += num_lines
+        f2["_size"] = file["_size"] - _file_size(f1)
```

Comment:
Doesn't this mean we're storing `file` in memory 3x's now?  Why not dereference `file` or just use it instead of copying?

```
f1 = file.copy()
file = None
f2 = f1.copy()
f1["_end"] = f1["_start"] + num_lines
f2["_start"] += num_lines
f2["offset"] += num_lines
f2["_size"] = f2["_size"] - _file_size(f1)
```

Reply:
Won't make any difference, since the only field that weighs anything is the `content` field, and it is copied by reference (`id(f1["content"]) == id(f2["content"])`) so no extra memory is alloced. Also assigning `file = None` won't GC anything since f1 still holds a reference to the content list.

Reply:
Ahh, I see.  I was thinking `copy()` was `deepcopy()`